src/site/docs/synonyms/assets/es6-samples.xml

```diff
@@ -0,0 +1,1038 @@
+  <synonym id="syn-for-in-loops">
+    <title>For-in loops</title>
+    <code language="javascript"><![CDATA[
+var fruits = ['orange', 'apple', 'banana'];
+
+// 'in' notation in JavaScript iterates through
+// the properties, not values, of an object. 
+// Thus it returns the indices of the array
+// and any items add to the Array prototype.
+
+for (var i in fruits) {
+  console.log(fruits[i]);
+}
+
+// You should use 'of' notation to iterates over
+// the values
+for (var fruit of fruits) {
+  console.log(fruit);
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-for-loops-for-objectsmaps">
+    <title>For loops for objects/maps</title>
+    <code language="javascript"><![CDATA[
+var data = { … };
+
+for (var key in data) {
+  console.log('key', key);
+  console.log('value', data[key]);
+}
+
+// Alternatively, the forEach loop is a
+// method on a Map in EcmaEscript6.
+var mymap = new Map();
+
+mymap.set('a', 1);
+mymap.set('b', 2);
+
+mymap.forEach(function(v, k, m) { console.log(v)});
+
+for(var m of mymap) {
+  console.log(m);
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-closures-and-counters-in-loops">
+    <title>Closures and counters in loops</title>
+    <code language="javascript"><![CDATA[
+var callbacks = [];
+
+// A closure must be used to preserve the return for
+// each function at each step of the loop.  Otherwise
+// every entry in callbacks will return 2;
+for (var i = 0; i < 2; i++) {
+  (function(_i) {
+    callbacks.push(function() {
+      return _i;
+    });
+  })(i);
+}
+
+// Without the internal closure, the result is 2
+callbacks[0]() // == 0
+
+// ECMAScript 6 can support this with the use of blocks
+let callbacks = [];
+for (let i = 0; i < 10; i++) {
+  let j = i;
+  callbacks.push(function() { console.log(j) });
+}
+
+callbacks[0]() // == 0
+]]></code>
+  </synonym>
+</theme>
+<theme id="theme-classes">
+  <title>Classes</title>
+  <synonym id="syn-define">
+    <title>Define</title>
+    <code language="javascript"><![CDATA[
+class Person {
+  constructor() {
+    this.name = null;
+  }
+
+  greet() {
+    return 'Hello, ' + this.name;
+  }
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-constructor-with-parameter">
+    <title>Constructor with parameter</title>
+    <code language="javascript"><![CDATA[
+class Person {
+  constructor(name) {
+    this.name = name;
+  }
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-instantiate">
+    <title>Instantiate</title>
+    <code language="javascript"><![CDATA[
+var person = new Person();
+]]></code>
+  </synonym>
+  <synonym id="syn-reflection">
+    <title>Reflection</title>
+    <code language="javascript"><![CDATA[
+var name = 'Bob';
+typeof name // == 'String'
+]]></code>
+  </synonym>
+  <synonym id="syn-check-the-type">
+    <title>Check the type</title>
+    <code language="javascript"><![CDATA[
+var name = 'Bob';
+
+// Because of javascript's type system we
+// need to check both instanceof and typeof
+name instanceof String || typeof name === 'string'; 
```

Comment:
I know this is from the original (https://www.dartlang.org/docs/synonyms/) but I don't really agree with this comment.